File: Math Problems/weirdGrid.py
import math
import os
import random
import re
import sys

# find number at row r and column c 
def strangeGrid(r, c):
    rowStart = (r - 1) // 2 * 10    # do integer division 
    if (r - 1) % 2 == 1:
        rowStart += 1
    return (rowStart + (c - 1) * 2)

if __name__ == '__main__':
    first_multiple_input = input().rstrip().split()
    r = int(first_multiple_input[0])
    c = int(first_multiple_input[1])
    result = strangeGrid(r, c)
    print(str(result) + '\n')

File: Math Problems/weirdGrid.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {
    
    // only need to find the values in row R to decrease runtime
    public static long strangeGrid(int r, int c) { 
        long rowStart = (r - 1) / 2;
        rowStart *= 10; 
        if ((r - 1) % 2 == 1) {
            rowStart += 1; 
        }
        
        return (rowStart + (c - 1) * 2);
    }
}

public class weirdGrid {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String[] firstMultipleInput = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        int r = Integer.parseInt(firstMultipleInput[0]);

        int c = Integer.parseInt(firstMultipleInput[1]);

        long result = Result.strangeGrid(r, c);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
